src/opi/input/structures/properties.py

```diff
@@ -0,0 +1,321 @@
+
+        # > converting into Path
+        trj_file = Path(trj_file)
+
+        # > Check if file exists
+        if not trj_file.exists():
+            raise FileNotFoundError(f"XYZ file not found: {trj_file}")
+
+        with trj_file.open() as f_xyz:
+            tracked = TrackingTextIO(f_xyz)
+            n_struc: int = 0
+            while True:
+                try:
+                    properties = cls.from_xyz_buffer(
+                        tracked, comment_symbols=comment_symbols, mode=mode
+                    )
+                    if properties is None:
+                        break
+                    properties_list.append(properties)
+                except ValueError:
+                    raise
+                n_struc += 1
+                # > Break if limit of allowed structures is reached
+                if struc_limit is not None and n_struc >= struc_limit:
+                    break
+        return properties_list
+
+    @classmethod
+    def from_xyz_block(
+        cls, xyz_string: str, mode: Literal["goat", "docker"] = "goat"
+    ) -> "Properties":
+        """
+        Function for reading xyz data from string and return a Properties object.
+
+        Parameters
+        ----------
+        xyz_string: str
+            String that contains xyz file data
+
+        Raises
+        --------
+        ValueError
+            If there is a problem with parsing the XYZ file
+
+        Returns
+        --------
+        Properties
+            The `Properties` object extracted from file
+        """
+        properties_list = cls.from_trj_xyz_block(xyz_string, struc_limit=1, mode=mode)
+        return properties_list[0]
+
+    @classmethod
+    def from_trj_xyz_block(
+        cls,
+        trj_string: str,
+        /,
+        *,
+        mode: Literal["goat", "docker"] = "goat",
+        comment_symbols: str | Sequence[str] | None = None,
+        struc_limit: int | None = None,
+    ) -> "list[Properties]":
+        """
+        Function for reading multi-xyz data from string and returning a Properties object.
+
+        Parameters
+        ----------
+        trj_string : Path | str | PathLike[str]
+            String that contains multiple xyz file data (trajectory data)
+        comment_symbols: str | Sequence[str] | None, default: None
+            List of symbols that indicate user comments in the xyz file. User comments are skipped before the actual xyz
+            data starts. By default, no user comments are used. White-space only comments are not allowed and are
+            silently ignored.
+        struc_limit: int | None, default: None
+            Limit of structures that should be read from the trj.xyz string. With the default, None, all structures are
+            read.
+
+        Returns
+        --------
+        `list[Properties]`:`Properties objects extracted from file
+        """
+        properties_list: list[Properties] = []
+
+        with StringIO(trj_string) as f_xyz:
```

Comment:
```suggestion
        with TrackingTextIO(trj_string) as f_xyz:
```
would that also be possible? If yes, please change.

Reply:
It is possible after implementing __enter__ and __exit__ to `TrackingTextIO` and allowing string initialization. I will change this accordingly.